### PR TITLE
Feat : (Add api routes add-project/add-user )

### LINKED ISSUE #116: [FEATURE] Adding API routes
### Description

for me right now we have to make API routes for - 
1. Adding a user 
2. Creating a project

### PR DESCRIPTION
**Pull request by AdityaPainuli**
### Added API routes for adding a project and for deleting a project

Closes #116 

### Changes proposed

I think we still need to work on the project schema there is some point we have to get fixed one of them is tags I am thinking of adding a many-to-many relationship so that we can use tags also as a search filter.
Talked about it on #116

### CODE REVIEW

**Comment by RakeshSangem** [comment]
Could you check what causing deployment failure!

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by AdityaPainuli** [comment]
image
It's working on the local environment @rohitdasu can you look over and tell me what is the error it's showing in vercel

File: pages/api/add-user.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    await prisma.user
+      .create({
+        data: {
+          email: email,
+        },
+      })
+      .then(() => {
+        res
+          .status(200)
+          .json({ message: 'User is created', success: true, data: null });
```
Comment (rohitdasu) [inline pages/api/add-user.ts:21]: you shouldn't return null

File: pages/api/add-user.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      })
+      .catch((e: any) => {
```
Comment (rohitdasu) [inline pages/api/add-user.ts:23]: don't use any

Comment (AdityaPainuli) [reply pages/api/add-user.ts:23]: got it 👍

File: pages/api/add-user.ts (hunk 1 above)
Comment (AdityaPainuli) [reply pages/api/add-user.ts:21]: then what should we written in data??

Comment (rohitdasu) [reply pages/api/add-user.ts:21]: you should return the data you saved. @leewan-09 do you have better suggestion then please share

Comment (AdityaPainuli) [reply pages/api/add-user.ts:21]: It's can't be return as it is a void function I tried to do that way .

Comment (rohitdasu) [reply pages/api/add-user.ts:21]: why void function?

Comment (AdityaPainuli) [reply pages/api/add-user.ts:21]: may be I am awaiting it. I don't know exactly but I get an error while doing that.

File: interfaces/index.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export type ResponseType = {
-  data: [] | {} | null;
+  data: [] | null | void;
```
Comment (rohitdasu) [inline interfaces/index.ts:2]: why void?

File: pages/api/add-project.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  tags: string[];
+  userId: string;
+};
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ResponseType>
+) {
+  const { author, title, description, githubRepo, tags, userId }: ProjectType =
+    req.body;
```
Comment (rohitdasu) [inline pages/api/add-project.ts:19]: can you put this inside the if statement pls

File: pages/api/add-project.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  if (req.method === 'POST') {
+    await prisma.project
+      .create({
+        data: {
+          author: author,
+          title: title,
+          description: description,
+          githubRepository: githubRepo,
+          userId: userId,
+          tags: tags,
```
Comment (rohitdasu) [inline pages/api/add-project.ts:29]:
```suggestion
          author,
          title,
          description,
          githubRepository: githubRepo,
          userId,
          tags,
```

File: pages/api/add-project.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        },
+      })
+      .then(() => {
+        res
+          .status(200)
+          .json({ message: 'User is created', success: true, data: null });
```
Comment (rohitdasu) [inline pages/api/add-project.ts:35]: in the data send the user information.

File: pages/api/add-user.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import { prisma } from '../../lib/prisma';
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ResponseType>
+) {
+  const { email } = req.body;
+
+  if (req.method === 'POST') {
+    const new_user = await prisma.user
```
Comment (rohitdasu) [inline pages/api/add-user.ts:12]: make the variable name newUser. please follow camel case in variables. in all places please refactor it to camel case

File: pages/api/add-user.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+      .create({
+        data: {
+          email: email,
```
Comment (rohitdasu) [inline pages/api/add-user.ts:15]:
```suggestion
          email,
```

File: pages/api/add-user.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        },
+      })
+      .then(() => {
+        res
+          .status(200)
+          .json({ message: 'User is created', success: true, data: new_user });
+      })
+      .catch((e) => {
+        console.log(e);
+        res.json({ data: null, message: e, success: false });
```
Comment (rohitdasu) [inline pages/api/add-user.ts:25]: no null please

File: interfaces/index.ts (hunk 3 above)
Comment (AdityaPainuli) [reply interfaces/index.ts:2]: because we are returning a void data  in new_data

Comment (rohitdasu) [reply interfaces/index.ts:2]: data should not return void in success cases. it should return something right

Comment (AdityaPainuli) [reply interfaces/index.ts:2]: but it's void type in our code

Comment (rohitdasu) [reply interfaces/index.ts:2]: could you please share the code which is void type in our code

Comment (AdityaPainuli) [reply interfaces/index.ts:2]: image
new_user is of void type

Comment (rohitdasu) [reply interfaces/index.ts:2]: you shouldn't directly return the new_user like this. see the response you are getting from the create function. and put that response inside the data. and also please make the variable name in camelCase.

**Review by leewan-09 — COMMENTED** [review]
If it working approve @rohitdasu. I'll like to do some more refactoring after the approval.

File: pages/api/add-project.ts [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    req.body;
+  if (req.method === 'POST') {
+    await prisma.project
+      .create({
+        data: {
+          author: author,
+          title: title,
+          description: description,
+          githubRepository: githubRepo,
+          userId: userId,
```
[the remark this answers is not in the document]
Comment (rohitdasu) [reply pages/api/add-project.ts]: yes now working

[the remark this answers is not in the document]
Comment (AdityaPainuli) [reply pages/api/add-project.ts]: it can work but if we have same name we can send like at presently.

**Review by rohitdasu — APPROVED** [review]
LGTM

**Pull request merged**